Automattic/simplenote-macos
# Manually update Sparkle to support Apple Silicon

In #803, we discussed using CocoaPods to fetch Sparkle. That's not possible because we need support for Sandboxing and Notarization and the version distributed via CocoaPods doesn't include it.

To support Sandboxing and Notarization, we need to use the WIP version 2.x branch. Unfortunately, the only way to use that version of Sparkle is to fetch it and build it manually (see `./Scripts/update-sparkle.sh`).


We cannot do like we've done many times with other libraries and point the `Podfile` to the the `2.x` branch because there the Pods [depends on a vendored framework](https://github.com/sparkle-project/Sparkle/blob/4767ef60ece7de7404f402c23177f0810bf72418/Sparkle.podspec#L27) which needs to [be built](https://github.com/sparkle-project/Sparkle/blob/4767ef60ece7de7404f402c23177f0810bf72418/Makefile#L16) every time. At least, that's my current understanding. I did try to just switch the branch and CocoaPods didn't generate any framework or fetch any source file

![image](https://user-images.githubusercontent.com/1218433/106237390-d7a9c580-6252-11eb-92e0-804d56d4a9e6.png)


For future reference, the commit I checked out was `4767ef60ece7de7404f402c23177f0810bf72418` (this information is in the update commit, too).

I considered using a submodule and scripting the build process, but decided against it because it would add complexity to the Git setup for a relatively uncommon operation – we don't need to update Sparkle often.

Having said that, I still wrote a script to perform the update, because it's not as straightforward as a drag-and-drop.

For more background on Sparkle Sandboxing and Notarization, see:

- https://sparkle-project.org/documentation/sandboxing/
- https://github.com/Automattic/simplenote-macos/pull/383
- https://github.com/Automattic/simplenote-macos/pull/481


### Test

I tested this by:

1. Change the version (`config/Version.public.xcconfig`) to be `2.6 (42)`
2. Archive this branch
3. Submit it for notarization (Distribute > Developer ID > Upload)
4. Once notarized (leave the window open and check after a few minutes), export the `.app` and copied it to the `/Applications` folder
5. Launch Simplenote

The Sparkle update should appear a few moments after and it should be possible to update to version 2.7

### Review

Keen to hear what @jkmassel has to say.
Also @jleandroperez because you worked on using Sparkle 2.x, so let me know if my setup looks coherent with the original.

### Release
These changes do not require release notes.
